runners/google-cloud-dataflow-java/build.gradle

```diff
@@ -520,8 +519,7 @@ task validatesRunnerV2 {
     excludedTests: [
       'org.apache.beam.sdk.transforms.ReshuffleTest.testReshuffleWithTimestampsStreaming',
 
-      // TODO(https://github.com/apache/beam/issues/18592): respect ParDo lifecycle.
-      'org.apache.beam.sdk.transforms.ParDoLifecycleTest.testFnCallSequenceStateful',
```

Comment:
The change only involves legacy runner, while this excluded test is removed. Can we confirm this now works with runner v2? One can trigger PostCommit validates runner test by adding any change to https://github.com/apache/beam/blob/master/.github/trigger_files/beam_PostCommit_Java_ValidatesRunner_Dataflow_V2.json and push a commit

Reply:
Triggering v1 and v2.  The  ExceptionThrowingFn in ParDoLifecycleTest only works if dofn execution is in the same process as the pipeline.run().  Should we add some new category of tests that require that so we can mark the tests with that instead of the exclusion list?  That could help other runners as well and if we modify the test and remove the label they would automatically start to run.  Thoughts?

Reply:
test passed.

> Should we add some new category of tests that require that so we can mark the tests with that instead of the exclusion list?

Yeah sounds like a better bookkeeping strategy, feel free to do though not blocking this PR, resolving for now.